### ISSUE TITLE
Function name must be a string

### ISSUE THREAD

**Issue by DavidSprauel** [dompdf/dompdf#2648]
I got this weird error that pop now since today (I might have updated the package tho).

```
  Function name must be a string

  at vendor/dompdf/dompdf/src/FontMetrics.php:136
    132▕         if (!defined("DOMPDF_FONT_DIR")) { define("DOMPDF_FONT_DIR", $fontDir); }
    133▕
    134▕         $file = $rootDir . "/lib/fonts/dompdf_font_family_cache.dist.php";
    135▕         $distFontsClosure = require $file;
  ➜ 136▕         $distFonts = $distFontsClosure($rootDir);
    137▕
    138▕         if (!is_readable($this->getCacheFile())) {
    139▕             $this->fontLookup = $distFonts;
    140▕             return;
```
    
 I check where in my code it trigger the error and seems to be there 

```php
 PDF::setOptions([
            'logOutputFile'   => storage_path('logs/log.htm'),
            'tempDir'         => storage_path('logs/'),
            'isRemoteEnabled' => true
        ]);
 ```

Everything was working smoothly before today and I detected the issue while launching my unit test. any idea ?
After checking what `$distFontsClosure` it seems to contains an array

**Comment by bsweeney** [comment]
We made a change to the format of the file that stores installed font information. I haven't run across this in local testing so I'll need to look into it more.

Is the dompdf_font_family_cache.dist.php file accessible to Dompdf?

**Comment by bsweeney** [comment]
Also, what framework are you using? Might be something about the specific implementation there.

**Comment by bsweeney** [comment]
If you wrap `$distFontsClosure` on line 136 in parenthesis does the error go away?
```
$distFonts = ($distFontsClosure)($rootDir);
```

**Comment by Mellthas** [comment]
Actually, #2421 mentions moving the `lib/fonts` directory, but it is not clear to me why that would be necessary.

**Comment by bsweeney** [comment]
I wonder if we should have released that change under the 2.0.0 release. Was hoping to avoid breaking changes but it looks like the change in format of the dompdf_font_family_cache.dist.php did, in fact, introduce the possibility of a breaking change.

Still, I haven't been able to reproduce myself with limited testing. The error I've gotten so far is much worse.

Perhaps we can mitigate the issue using the same logic we use for the generated file:

```
$distFonts = is_array($distFontsClosure) ? $distFontsClosure : $distFontsClosure($rootDir);
```

**Comment by Mellthas** [comment]
Steps to reproduce:
1. Set the `rootDir` option to a directory other than the dompdf installation directory.
2. Copy `lib/fonts` and `lib/res` from an older dompdf and put them into that directory.

**Comment by DavidSprauel** [comment]
Sorry for late answer. 

i'm using php7.4 and laravel framework, dompdf is used via larvel-dompdf package.
I've updated everything in my composer but the issue is still here. Have I missed something ?

**Comment by Mellthas** [comment]
The new release is not out yet. You should also be able to fix the issue by copying `lib/fonts/dompdf_font_family_cache.dist.php` from the dompdf directory to the directory set by your `rootDir` option, so that the new version of the file is used.

If you haven’t modified the `rootDir` option, then the issue must be caused by something else. Can you tell us more about your setup in that case?

**Comment by bsweeney** [comment]
Just released the 1.1.1 version. It should address this issue.

**Comment by DavidSprauel** [comment]
Problem solved with the 1.1.1 ^^thanks guys

**Comment by DavidSprauel** [comment]
This was this only thing I was modifying:
```php
 PDF::setOptions([
            'logOutputFile'   => storage_path('logs/log.htm'),
            'tempDir'         => storage_path('logs/'),
            'isRemoteEnabled' => true
        ]);
```
